src/main/webapp/dev/js/main/feedbackSubmissionsEdit.js

```diff
@@ -1291,6 +1291,100 @@ function updateTextQuestionWordsCount(textAreaId, wordsCountId, recommendedLengt
+    const isShowSectionTeamChecked = $(`#showSectionTeamFlag-${qnNum}`).prop('checked');
+
+    function nameComparator(e1, e2) {
+        const e1Name = e1.text;
+        const e2Name = e2.text;
+
+        let result = 0;
+        if (e1Name > e2Name) {
+            result = 1;
+        } else if (e1Name < e2Name) {
+            result = -1;
+        }
+        return result;
+    }
+
+    function sectionTeamComparator(e1, e2) {
+        // empty dropdowns should always appear at the top
+        if (e1.text === '') {
+            return -1;
+        }
+        if (e2.text === '') {
+            return 1;
+        }
+        const e1Section = $(e1).attr('section');
+        const e2Section = $(e2).attr('section');
+
+        const e1Team = $(e1).attr('team');
+        const e2Team = $(e2).attr('team');
+
+        const e1Name = $(e1).attr('name');
+        const e2Name = $(e2).attr('name');
+
+        // sort based on lexicographical ordering of section followed by team and name
+        let result = 0;
+        if (e1Section > e2Section) {
+            result = 1;
+        } else if (e1Section < e2Section) {
+            result = -1;
+        } else if (e1Team > e2Team) {
+            result = 1;
+        } else if (e1Team < e2Team) {
+            result = -1;
+        } else if (e1Name > e2Name) {
+            result = 1;
+        } else if (e1Name < e2Name) {
+            result = -1;
+        }
+        return result;
+    }
+
+    // iterate through each dropdown list and add/remove section and team fields
+    for (let i = 0; i < numResponses; i += 1) {
+        const evaluees = $(`select[name='responserecipient-${qnNum}-${i}'] option`);
+        const selectedOption = $(`select[name='responserecipient-${qnNum}-${i}']`).val();
+        if (isShowSectionTeamChecked) {
+            // iterate through each option value in dropdown and augment it with section and team
+            $(evaluees).each(function () {
+                const evalueeSection = $(this).attr('section');
+                const evalueeTeam = $(this).attr('team');
+                const evalueeName = $(this).text();
+                // leave empty option as it is
+                if (evalueeName === '') {
+                    return;
+                }
+                $(this).text(`${evalueeSection} | ${evalueeTeam} | ${evalueeName}`);
+            });
+            evaluees.sort(sectionTeamComparator);
+        } else {
+            // set each option value as evaluee's name
+            $(evaluees).each(function () {
+                // leave empty option as it is
+                if ($(this).text() === '') {
+                    return;
+                }
+                const evalueeName = $(this).attr('name');
+                $(this).text(evalueeName);
+            });
+            evaluees.sort(nameComparator);
+        }
+
+        // refresh dropdown list
+        $(`select[name='responserecipient-${qnNum}-${i}']`).empty().append(evaluees);
```

Comment:
Can you try `detach` instead of `empty`? maybe that will keep the selected option so that you don't need the next line anymore.

Reply:
`detach` removes the `select` element itself as well which means we'll have to reinsert it as well along with the options. Is it worth it ?